manticore/ethereum.py

```diff
@@ -356,24 +356,26 @@ def __build_source_map(self, bytecode, srcmap):
         f = int(md.get(2, 0))
         j = md.get(3, None)
 
+        pos_to_offset = {}
         for i in evm.EVMAsm.disassemble_all(bytecode):
-            if asm_pos in srcmap and len(srcmap[asm_pos]):
-                md = srcmap[asm_pos]
-                if len(md):
-                    d = {}
-                    for p, k in enumerate(md.split(':')):
-                        if len(k):
-                            d[p] = k
-
-                    s = int(d.get(0, s))
-                    l = int(d.get(1, l))
-                    f = int(d.get(2, f))
-                    j = d.get(3, j)
-
-            new_srcmap[asm_offset] = (s, l, f, j)
+            pos_to_offset[asm_pos] = asm_offset
             asm_pos += 1
             asm_offset += i.size
 
+        for asm_pos, md in enumerate(srcmap):
+            if len(md):
+                d = {}
+                for p, k in enumerate(md.split(':')):
+                    if len(k):
+                        d[p] = k
+
+                s = int(d.get(0, s))
+                l = int(d.get(1, l))
+                f = int(d.get(2, f))
+                j = d.get(3, j)
```

Comment:
can we rename those to sth more explicit? ;/

Reply:
Yes captain!